### DISCUSSION TITLE
GraalVM native images

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#318] [Ideas]
@bevans2000 @Urwumpe 

I must say it's quite an interesting proposition to make mars-sim executable. 

Since it relies on Java VM, we ask players to download and install Java 11 on their machine.

When it comes to installing Java, there's still all kinds of issues, especially if you already have older versions of Java and the settings already got the PATH and JAVA_HOME variable messed up, etc.

Wouldn't it be better if a lean Java VM is bundled with every mars-sim release ?

Anyway, I used to play with creating GraalVM native images here and there.

Though haven't been successful with mars-sim, it's worth our effort looking into this approach in near future.

See [this](https://medium.com/graalvm/graalvm-native-image-quick-reference-4ceb84560fd8)

Does it sound like a good idea to try out ?

**Comment by Urwumpe** [comment]
Well, we could try this and take a look how many problems we get in the attempt, make issues with a "GraalVM" label for documenting the problems. And then decide if and how we fix those problems. I agree that a "run anywhere" package might be appealing, but it would also add a new dependency and a new technology to support. If there are restrictions, we should know them in advance. 

Please note, that the Native Image feature of GraalVM requires enterprise subscription, for me personally, that would mean its a no-go to make it default distribution or include it in our public build pipeline. Should this change, no problem to talk about it again. 

https://www.graalvm.org/faq/

Another option for us would be downgrading everything to JDK8 compatibility. But I think installing a JDK is a fair price for the user.
